common/txsummary.go

```diff
@@ -0,0 +1,114 @@
+package common
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+var TxSummaryEntryCSVHeader = []string{
+	"timestamp_ms",
+	"hash",
+	"chain_id",
+	"from",
+	"to",
+	"value",
+	"nonce",
+	"gas",
+	"gas_price",
+	"gas_tip_cap",
+	"gas_fee_cap",
+	"data_size",
+	"data_4bytes",
+	"sources",
+	"included_at_block_height",
+	"included_block_timestamp_ms",
+	"inclusion_delay_ms",
+}
+
+// TxSummaryEntry is a struct that represents a single transaction in the summary CSV and Parquet file
+// see also https://github.com/xitongsys/parquet-go for more details on parquet tags
+type TxSummaryEntry struct {
+	// The fields are written to CSV, and the order shouldn't change (for backwards compatibility)
+	Timestamp int64  `parquet:"name=timestamp, type=INT64, convertedtype=TIMESTAMP_MILLIS"`
+	Hash      string `parquet:"name=hash, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+
+	ChainID string `parquet:"name=chainId, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN"`
+	From    string `parquet:"name=from, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+	To      string `parquet:"name=to, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN_DICTIONARY"`
+	Value   string `parquet:"name=value, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+	Nonce   string `parquet:"name=nonce, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+
+	Gas       string `parquet:"name=gas, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+	GasPrice  string `parquet:"name=gasPrice, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+	GasTipCap string `parquet:"name=gasTipCap, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+	GasFeeCap string `parquet:"name=gasFeeCap, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN, omitstats=true"`
+
+	DataSize   int64  `parquet:"name=dataSize, type=INT64"`
+	Data4Bytes string `parquet:"name=data4Bytes, type=BYTE_ARRAY, convertedtype=UTF8, encoding=PLAIN_DICTIONARY"`
+
+	Sources []string `parquet:"name=sources, type=MAP, convertedtype=LIST, valuetype=BYTE_ARRAY, valueconvertedtype=UTF8"`
+
+	// Inclusion stats
+	IncludedAtBlockHeight  int64 `parquet:"name=includedAtBlockHeight, type=INT64"`
+	IncludedBlockTimestamp int64 `parquet:"name=includedBlockTimestamp, type=INT64, convertedtype=TIMESTAMP_MILLIS"`
+	InclusionDelayMs       int64 `parquet:"name=inclusionDelayMs, type=INT64"`
```

Comment:
tbd whether this is desirable

Reply:
valuable, one less thing in between researchers and insights imo🚀